src/custom/utils/tokens.ts

```diff
@@ -9,3 +10,12 @@ export function toErc20Address(tokenAddress: string, chainId: ChainId): string {
 
   return checkedAddress
 }
+
+export function toNativeBuyAddress(tokenAddress: string, chainId: ChainId): string {
```

Comment:
the whole function is a C&P, `toErc20Address`

I think it can be refactored. At least we could do the `isNativeCurrency(address): boolean`

Reply:
Yes, that's obvious. Felt more explicit this way

Reply:
but will refactor if it's blocking this hotfix